js/stdlib/ts/shared_backend.ts

```diff
@@ -304,3 +304,9 @@ export const simTokenBurn = (sim_r:any, tok:any, amt:any): void => {
 export const simTokenDestroy = (sim_r:any, tok:any): void => {
   sim_r.txns.push({kind: 'tokenDestroy', tok});
 };
+
+export const simTokenAccepted = (sim_r:any, addr:any, tok:any): boolean => {
+  sim_r.txns.push({kind: 'tokenAccepted', addr, tok});
+  // TODO - I need to query for acceptance here, but that requires things from ALGO.ts...
```

Comment:
The internals of `acc.tokenAccepted` don't require an account object, but they do require a provider and any way I slice it I need to either get stuff from `ALGO.ts` or re-implement bigger things than I would like to.  `shared_backend.ts` says it has no dependencies on other shared things (although it does import some stuff...).  How do you feel about moving code from `ALGO.ts` into `shared_impl`?

Reply:
You should add a function to `IContractCompiled` --- https://github.com/reach-sh/reach-lang/blob/master/js/stdlib/ts/shared_impl.ts#L187 --- and generate a call like `ctc.simTokenAccepted` and it will call `tokenAcceptedThingy` and pass it the provider of the account that created the `etc` handle